### PR TITLE
refactor: migration of nuxt route middleware  

### PR DESCRIPTION
**Pull request by Jarsen136**
## PR Type

- [x] Bugfix
- [x] Refactoring

## Context

- [x] Related to #7034
- [x] migration of nuxt route middleware  
ref: https://nuxt.com/docs/guide/directory-structure/middleware

### CODE REVIEW

File: components/carousel/utils/useCarouselSpotlight.ts [hunk 1]
```diff
@@ -61,12 +61,11 @@ function updateCollections(data) {
 
 export default function useCarouselSpotlight() {
   const collections = useSpotlightNft()
-  const { urlPrefix, client } = usePrefix()
+  const { urlPrefix } = usePrefix()
   const variables = curatedCollection[urlPrefix.value]?.length
     ? { list: curatedCollection[urlPrefix.value] }
     : undefined
   const { data } = useGraphql({
-    clientName: client,
```
Comment (roiLeo) [inline components/carousel/utils/useCarouselSpotlight.ts:69]: throw warning without, I think we need to have default value for `clientName` in `useGraphql`

**Review by roiLeo — APPROVED** [review]

File: components/carousel/utils/useCarouselSpotlight.ts (hunk 1 above)
Comment (Jarsen136) [reply components/carousel/utils/useCarouselSpotlight.ts:69]: It's just a type warning. I will fix it.

File: composables/useGraphql.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
 import resolveQueryPath from '@/utils/queryPathResolver'
 import { notificationTypes, showNotification } from '@/utils/notification'
 import type { QueryOptions } from 'apollo-client'
-
+import type { ComputedRef } from 'vue'
```
Comment (roiLeo) [inline composables/useGraphql.ts:4]:
```suggestion
import { ComputedRef } from 'vue/types'
```

**Review by roiLeo — APPROVED** [review]

File: components/carousel/utils/useCarouselSpotlight.ts (hunk 1 above)
Comment (roiLeo) [reply components/carousel/utils/useCarouselSpotlight.ts:69]: Still show warning for me:

```
Argument of type '{ queryName: string; queryPrefix: Prefix; variables: { list: any; } | undefined; }' is not assignable to parameter of type '{ queryPrefix?: string | undefined; queryName: any; clientName: any; variables?: {} | undefined; options?: {} | undefined; disabled?: ComputedRef<boolean> | undefined; data?: Ref<...> | undefined; error?: Ref<...> | undefined; loading?: Ref<...> | undefined; }'.
  Property 'clientName' is missing in type '{ queryName: string; queryPrefix: Prefix; variables: { list: any; } | undefined; }' but required in type '{ queryPrefix?: string | undefined; queryName: any; clientName: any; variables?: {} | undefined; options?: {} | undefined; disabled?: ComputedRef<boolean> | undefined; data?: Ref<...> | undefined; error?: Ref<...> | undefined; loading?: Ref<...> | undefined; }'.
```

Comment (roiLeo) [reply components/carousel/utils/useCarouselSpotlight.ts:69]: works with https://github.com/kodadot/nft-gallery/pull/7208#discussion_r1319546597 this change

Comment (Jarsen136) [reply components/carousel/utils/useCarouselSpotlight.ts:69]: thank you for your update

**Pull request merged**